technologies/job/knime/login-auth/instanceActions.js

```diff
@@ -0,0 +1,215 @@
+const axios = require('axios');
+const { Response, JobStatus, Log, Stream } = require('@saagie/sdk');
+const fs = require('fs');
+const extract = require('extract-zip');
+const rimraf = require('rimraf');
+const dayjs =  require('dayjs');
+const { JOB_STATES } = require('../job-states');
+const { getRequestConfigFromEndpointForm } = require('./utils');
+const { ERRORS_MESSAGES } = require('../errors');
+
+const getLogsForAJob = (logsLines, lineIndex, jobIdString) => {
+  let currentLineIndex = lineIndex;
+  let logs = [];
+
+  if (logsLines && logsLines.length > currentLineIndex) {
+    // We are detecting if the current line is the first of a log block (start by a date) for the selected job
+    const maybeDate = logsLines[currentLineIndex].substring(0, 19);
+    let isAValidDate = dayjs(maybeDate).isValid();
+
+    if (isAValidDate && logsLines[currentLineIndex].includes(jobIdString)) {
+      // We add the first log block for the selected job
+      logs.push(logsLines[currentLineIndex]);
+
+      do {
+        // We add log block lines until to find a new log block (next line to start by a date)
+        if (logsLines.length > currentLineIndex + 1) {
+          const maybeDate = logsLines[currentLineIndex + 1].substring(0, 19);
+          isAValidDate = dayjs(maybeDate).isValid();
+          if (!isAValidDate) {
+            logs.push(logsLines[currentLineIndex + 1]);
+            currentLineIndex++;
+          }
+        }
+      } while (!isAValidDate && logsLines.length > currentLineIndex + 1);
+    }
+
+    if (logsLines && logsLines.length > currentLineIndex + 1) {
+      // We go to the next line to find a new log block for the selected job
+      return [
+        ...logs,
+        ...getLogsForAJob(logsLines, currentLineIndex + 1, jobIdString),
+      ];
+    }
+  }
+
+  return logs;
+};
+
+/**
+ * Logic to start the external job instance.
+ * @param {Object} params
+ * @param {Object} params.job - Contains job data including featuresValues.
+ * @param {Object} params.instance - Contains instance data.
+ */
+exports.start = async ({ job, instance }) => {
+  try {
+    console.log('START INSTANCE:', instance);
+
+    let jobId = job.featuresValues.job.id;
+
+    if (job.featuresValues.job.id === 'run-new-job') {
```

Comment:
'run-new-job' can probably be defined as a constant as it is redefined in the other file :wink: